sdk/storage_queues/src/clients/pop_receipt_client.rs

```diff
@@ -56,16 +64,19 @@ impl PopReceiptClient {
         Ok(url)
     }
 
-    /// Deletes the message. The message must not have been
-    /// made visible again or this call would fail.
+    /// Deletes the message. The message must not have been made visible again
+    /// or this call would fail.
     pub fn delete(&self) -> DeleteMessageBuilder {
-        DeleteMessageBuilder::new(self)
+        DeleteMessageBuilder::new(self.clone())
     }
 
-    /// Updates the message.
-    /// The message must not have been
-    /// made visible again or this call would fail.
-    pub fn update(&self, visibility_timeout: impl Into<VisibilityTimeout>) -> UpdateMessageBuilder {
-        UpdateMessageBuilder::new(self, visibility_timeout)
+    /// Updates the message.  The message must not have been made visible again
+    /// or this call would fail.
```

Comment:
```suggestion
    /// Updates the message.
    /// 
    /// The message must not have been made visible again or this call would fail.
```

Also, it might be nice to place this above `delete`. In Cosmos, we generally follow the order: get, create, update, delete for ordering operations.